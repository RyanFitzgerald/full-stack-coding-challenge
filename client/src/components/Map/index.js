import React from 'react';
import PropTypes from 'prop-types';
import { withScriptjs, withGoogleMap, GoogleMap } from "react-google-maps";

// Define the component
const Map = withScriptjs(withGoogleMap(function({ children }) { 
  return (
    <GoogleMap
      defaultZoom={4}
      defaultCenter={{ lat: 40.1950, lng: -95.1402 }}
    >
      {children}
    </GoogleMap>
  );
}));

// Define prop types
Map.propTypes = {
  children: PropTypes.any.isRequired
}

export default Map;