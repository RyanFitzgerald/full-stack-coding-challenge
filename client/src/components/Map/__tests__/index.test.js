import React from 'react';

import Map from '../index';

describe('<ResultMap />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(
      <Map
        googleMapURL="https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places"
        loadingElement={<div style={{ height: `100%` }} />}
        containerElement={<div style={{ height: `400px` }} />}
        mapElement={<div style={{ height: `100%` }} />}
      />
    ).toJSON();
    
    expect(renderedComponent).toMatchSnapshot();
  });
});