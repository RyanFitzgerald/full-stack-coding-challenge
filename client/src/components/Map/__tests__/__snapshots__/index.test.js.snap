// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultMap /> should match the snapshot 1`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
/>
`;
